app/uk/gov/hmrc/transitmovementsrouter/connectors/PersistenceConnector.scala

```diff
@@ -77,8 +78,9 @@ class PersistenceConnectorImpl @Inject() (httpClientV2: HttpClientV2, appConfig:
         .withBody(source)
         .execute[Either[UpstreamErrorResponse, HttpResponse]]
         .map {
-          case Right(_) =>
-            Right(())
+          case Right(res) =>
+            val m = Json.toJson(res.body) \ "messageId"
+            Right(MessageId(m.head.toString))
```

Comment:
You wouldn't need to do this if there is a format or reads defined for MessageId

`  implicit lazy val messageIdFormat: Format[MessageId] = Json.valueFormat[MessageId]
` for example.

Reply:
Also you need to update the persistence connector spec